src/Network/NWConnectionGroup.cs

```diff
@@ -0,0 +1,276 @@
+
+		public NWPath? GetPath (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_copy_path_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWPath (ptr, owns: true);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_endpoint nw_connection_group_copy_remote_endpoint_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWEndpoint? GetRemmoteEndpoint (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_copy_remote_endpoint_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWEndpoint (ptr, owns: true);
+		}
+
+		// can return null
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_connection nw_connection_group_extract_connection_for_message (OS_nw_connection_group group, OS_nw_content_context context);
+
+		public NWConnection? GetConnection (NWContentContext context)
+		{
+			if (context == null)
+				throw new ArgumentNullException (nameof (context));
+			var ptr = nw_connection_group_extract_connection_for_message (GetCheckedHandle (), context.Handle);
+			return ptr == IntPtr.Zero ? null : new NWConnection (ptr, owns: true);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		static extern void nw_connection_group_reply (OS_nw_connection_group group, OS_nw_content_context inbound_message, OS_nw_content_context outbound_message, /* [NullAllowed]  DispatchData */ IntPtr content);
+
+		public void Reply (NWContentContext inboundMessage, NWContentContext outboundMessage, DispatchData content)
+		{
+			if (inboundMessage == null)
+				throw new ArgumentNullException (nameof (inboundMessage));
+			if (outboundMessage == null)
+				throw new ArgumentNullException (nameof (outboundMessage));
+
+			var contentPtr = (content == null) ? IntPtr.Zero : content.Handle;
+			nw_connection_group_reply (GetCheckedHandle (), inboundMessage.Handle, outboundMessage.Handle, contentPtr);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_connection_group_send_message (OS_nw_connection_group group, /* [NullAllowed] DispatchData */ IntPtr content, /* [NullAllowed] */ OS_nw_endpoint endpoint, OS_nw_content_context context, BlockLiteral *handler);
+
+		delegate void nw_connection_group_send_completion_t (IntPtr block, IntPtr error);
+		static nw_connection_group_send_completion_t static_SendCompletion = TrampolineSendCompletion;
+
+		[MonoPInvokeCallback (typeof (nw_connection_group_send_completion_t))]
+		static void TrampolineSendCompletion (IntPtr block, IntPtr error)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWError?>> (block);
+			if (del != null) {
+				var err = error == IntPtr.Zero ? null : new NWError (error, owns: false);
+				del (err);
+				err?.Dispose ();
```

Comment:
To my understanding, a question mark denotes a nullable type, but since you are not declaring 'del' here, why is there a question mark?

Reply:
In this case is because error can be null, the ? will mean, if not null, call the method, else don't